add CheckSemVer, CheckRootfsPath, CheckPlatform unit test

liangchenye opened the pull request:
Signed-off-by: liangchenye <liangchenye@huawei.com>

wking commented on .travis.yml:
```diff
@@ -14,5 +14,6 @@ script:
   - $HOME/gopath/bin/golint ./...
   - $HOME/gopath/bin/git-validation -run DCO,short-subject -v -range ${TRAVIS_COMMIT_RANGE}
+  - go test -v ./...
   - make
```
Probably better to drop the explicit `vet` and `golint` calls and have:

```
script:
    - $HOME/gopath/bin/git-validation -run DCO,short-subject -v -range ${TRAVIS_COMMIT_RANGE}
    - make all test
```

liangchenye replied:
how about  test first then try to compile?
- make test
- make all

wking replied:
That's fine with me.

wking commented on Makefile:
```diff
@@ -37,7 +37,7 @@ $(RUNTIME_TOOLS_LINK):
 
 .PHONY: test .gofmt .govet .golint
```
You need to add `.gotest` here as well.

liangchenye replied:
Yes, I miss this.

wking replied:
Still missing a `.PHONY` entry for `.gotest`?

liangchenye commented on validate/validate_test.go:
```diff
@@ -0,0 +1,82 @@
+		expected bool
+	}{
+		{rspec.Version, true},
+		//FIXME: validate currently only handles rpsec.Version
```
We don't have the 'official supported version', so I prefer to keep this  and update this once the problem of supporting past version is fixed.

liangchenye commented on validate/validate_test.go:
```diff
@@ -0,0 +1,82 @@
+	rootfsDir := "rootfs"
+	rootfsNonDir := "rootfsfile"
+	rootfsNonExists := "rootfsnil"
+	os.MkdirAll(filepath.Join(tmpBundle, rootfsDir), 0750)
```
I just used to set 0750 to all the created dir. I don't mind to change it to 0700. Update comes soon.

Mashimiao commented:
LGTM

liangchenye commented:
ping @mrunalp @hqhq

mrunalp commented on validate/validate_test.go:
```diff
@@ -0,0 +1,82 @@
+	rootfsNonDir := "rootfsfile"
+	rootfsNonExists := "rootfsnil"
+	os.MkdirAll(filepath.Join(tmpBundle, rootfsDir), 0700)
+	os.Create(filepath.Join(tmpBundle, rootfsNonDir))
```
Missing error check.

liangchenye replied:
fixed

liangchenye commented on validate/validate_test.go:
```diff
@@ -0,0 +1,82 @@
+	rootfsDir := "rootfs"
+	rootfsNonDir := "rootfsfile"
+	rootfsNonExists := "rootfsnil"
+	os.MkdirAll(filepath.Join(tmpBundle, rootfsDir), 0700)
```
fixed

liangchenye commented:
rebased, @mrunalp PTAL

Mashimiao commented:
LGTM

Mashimiao commented:
need rebase

liangchenye commented:
rebased, PTAL  @Mashimiao  @hqhq @mrunalp

Mashimiao commented:
failed with CI

liangchenye commented:
updated @Mashimiao 
Since we have validation/validation_test.go which is not UT, I added a UTDIRS to Makefile and now we will only do unit test under UTDIRS.

Mashimiao commented:
LGTM

hqhq commented on validate/validate_test.go:
```diff
@@ -0,0 +1,86 @@
+	}
+}
+
+func TestCheckPlatform(t *testing.T) {
```
Platform is already removed from runtime-spec, so we don't need to add this is we're aiming at 1.0.

liangchenye commented:
'Platform' UT is removed @hqhq 
I planed to remove the platform validation code this afternoon, but found it needs more time.
So I opened an issue to keep track of this. 
https://github.com/opencontainers/runtime-tools/issues/407

hqhq commented:
LGTM

Mashimiao commented:
LGTM
